include/up-transport-zenoh-cpp/ZenohUTransport.h

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_TRANSPORT_ZENOH_CPP_ZENOHUTRANSPORT_H
+#define UP_TRANSPORT_ZENOH_CPP_ZENOHUTRANSPORT_H
+
+#include <up-cpp/transport/UTransport.h>
+
+#include <filesystem>
+#include <optional>
+
+namespace uprotocol::transport {
+
+/// @brief Zenoh implementation of UTransport
+///
+/// This implementation must meet the following requirements:
+///
+/// * [MUST] An implementation MUST support multiple simultaneous instantiatons
+///          from within the same application.
+///
+/// * [MAY]  An implimentation MAY require that each new instance within an
+///          application have a unique configuration from existing instances.
+///
+/// * [MUST] An implementation MUST allow for multiple users of the same
+///          instance.
+///
+/// * [MUST] An implementation MUST be thread-safe.
+///
+/// * [MUST] Throw an exception if the transport fails to initialize or the
+///          configuration is invalid.
+struct ZenohUTransport : public UTransport {
+	/// @brief Constructor
+	///
+	/// @param defaultUri Default Authority and Entity (as a UUri) for
+	///                   clients using this transport instance.
+	/// @param configFile Path to a configuration file containing the Zenoh
+	///                   transport configuration.
+	ZenohUTransport(const v1::UUri& defaultUri,
+	                const std::filesystem::path& configFile);
```

Comment:
```suggestion
	ZenohUTransport(const v1::UUri& defaultUri,
	                const std::variant<std::filesystem::path, std::string>& configFile);
```

Consider switching to take _either_ a path _or_ a pre-loaded configuration file.

Reply:
IMHO giving too many options will just confuse people so I recommend picking on and sticking with it (better if we do the same for all transports)

Reply:
I would recommend using Config struct here. People can create their own struct or load it from path easily (also ensure flexibility).
This is also used by up-client-zenoh-rust, and I showed both ways in the example here.
https://github.com/eclipse-uprotocol/up-zenoh-example-rust/blob/633e889470e03d18ab493fcce54fd2e6751b36a7/src/common_uuri.rs#L65

Reply:
@evshary I'm not entirely sure I understand the code you linked to. Is it loading a json file and parsing it to a native object? We don't have a json parser currently in the C++ libraries, and it would be nice to avoid putting that burden on users. Additionally, it appears the Zenoh C++ API takes an open file object and not a data structure when creating a session.

I think the best option for us here is to go back to simply taking a file path. Then the implementation of this constructor can open that file and pass it to the Zenoh session object.

Reply:
No, I mean loading the object Config like this
https://github.com/eclipse-zenoh/zenoh-cpp/blob/main/examples/universal/z_put.cxx#L44
Users can decide whether to create the object by themselves or retrieve it from a file path.

Reply:
@evshary - I would argue that code should be _inside_ the ZenohUTransport implementation and not _outside_ it. The code you pointed to is still loading a configuration file, so the most generic interface we can provide outward to uProtocol users is a file path.

I do not thing it makes any sense for a caller to this constructor to need to know anything about a zenoh-cpp API for loading configuration files into objects.